platform/linux-generic/odp_ipsec.c

```diff
@@ -1884,9 +1885,6 @@ int odp_ipsec_out_inline(const odp_packet_t pkt_in[], int num_in,
 		in_pkt++;
 		sa_idx += sa_inc;
 		opt_idx += opt_inc;
-
-		/* Last thing */
-		_odp_ipsec_sa_unuse(ipsec_sa);
```

Comment:
I'm unsure about this patch. Application can not know when out inline packets were actually sent, so an application will send several packets using `odp_ipsec_out_inline()`, then it will call `odp_ipsec_sa_disable()` and wait for SA to be actually disabled. Thus refcounting is necessary here.

Reply:
After discussion, this code will be safe with the current code. Could you please add huuuge warning, that if one would like to add async crypto as IPsec backing code or proper async IPsec implementation, he has to re-implement some kind of SA safety here.

Reply:
@Bill-Fischofer-Linaro The speedup of a single patch in this patch set depends on the order of the patches. In the current patch order, the refcount removal patch speeds up my unidirectional IPsec tunneling test about 45 % with 10 worker threads.